jormungandr/src/settings/logging.rs

```diff
@@ -83,125 +75,138 @@ impl FromStr for LogOutput {
+                    LogFormat::Plain => Ok((Box::new(builder.finish()), Some(guard))),
+                    LogFormat::Json => Ok((Box::new(builder.json().finish()), Some(guard))),
                 }
             }
-            #[cfg(unix)]
-            LogOutput::SyslogUdp { host, hostname } => {
-                use std::net::{IpAddr, Ipv4Addr, SocketAddr};
-
-                format.require_plain()?;
-
-                let mut local_port = 30_000;
-                let host = host.parse().map_err(Error::SyslogInvalidHost)?;
-
-                // automatically select local port
-                loop {
-                    let local =
-                        SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), local_port);
-                    let res = slog_syslog::SyslogBuilder::new()
-                        .facility(Facility::LOG_USER)
-                        .udp(local, host, hostname)
-                        .start();
-                    match res {
-                        Ok(drain) => break Ok(drain.into_async()),
-                        Err(e) => {
-                            if e.kind() == std::io::ErrorKind::AddrInUse && local_port < 65_535 {
-                                local_port += 1;
-                                continue;
-                            }
-
-                            break Err(Error::SyslogAccessFailed(e));
-                        }
-                    }
+            LogOutput::Stderr => {
+                let (subscriber, guard) = tracing_appender::non_blocking(std::io::stderr());
+                let builder = builder.with_writer(subscriber).with_max_level(*level);
+                match format {
+                    LogFormat::Plain => Ok((Box::new(builder.finish()), Some(guard))),
+                    LogFormat::Json => Ok((Box::new(builder.json().finish()), Some(guard))),
                 }
             }
             #[cfg(feature = "systemd")]
             LogOutput::Journald => {
+                let layer = tracing_journald::layer()?;
                 format.require_plain()?;
-                Ok(JournaldDrain.into_async())
+                Ok((
+                    Box::new(builder.with_max_level(level).with_subscriber(layer)),
+                    None,
+                ))
             }
             #[cfg(feature = "gelf")]
             LogOutput::Gelf {
                 backend: graylog_host_port,
-                log_id: graylog_source,
+                log_id: _graylog_source,
             } => {
                 // Both currently recognized formats can be understood to apply:
                 // GELF formats payloads in JSON so 'json' is redundant,
                 // and plain messages are worked into JSON just the same.
                 // Match them irrefutably so that any new format will need to
                 // be addressed here when added.
-                match format {
-                    LogFormat::Plain | LogFormat::Json => {}
-                };
```

Comment:
The comment applied to the irrefutable match statement that was a guard against adding a format in the future that is not already supported by graylog. Please restore this.

Reply:
I wanted to ask about this when I first saw it but then forgot to did it in the middle of mayhem. 
If gelf already use its own format which we cannot change, why bother with the format anyway? Just in case we want to assert that configuration is "correct", otherwise ignoring it should be good enough.

Reply:
It's just that we have this format option that is not entirely orthogonal to the log backend.

Maybe it would be better to just ignore the format and warn that it has no effect when the backend is anything other than stdout/stderr/file.

Reply:
Added a "default" format that is chosen by default (naturally), means the same as "plain" on file descriptor appender backends, and is the only format allowed on journald and gelf.